### ISSUE TITLE
Feature: port python's hybrid search to rust

### ISSUE THREAD

**Issue by westonpace** [lancedb/lancedb#995]
### SDK

Rust

### Description

Python has a hybrid search API that should be ported to rust if we want to be able to use it in all of the SDKs

**Comment by changhiskhan** [comment]
unclear how we'll want to do this. because for reranking there are options that would not be available in rust (e.g., cross encoder / colbert) unless we also fetch the ONNX version. But at least cohere / linear combination should be doable.

**Comment by westonpace** [comment]
Yeah, same thoughts as #994 apply here.  Rust creates the extension point and the python implementation hooks the extension point into python-specific libraries to offer extensions like cohere and colbert.

**Comment by samlhuillier** [comment]
@westonpace would be very keen to see this in javascript! Do you have an ETA for when you think it'll be available?

**Comment by westonpace** [comment]
No ETA at the moment.  I'm currently trying to unify the functionality that is available on all three SDKs.  I'm hoping to finish that in the next week or two.  Porting SDK-specific functionality will come sometime after that.  Per @changhiskhan 's comment, were you looking for simple linear reranking or something more complex?

**Comment by samlhuillier** [comment]
Apologies for the abominable delay @westonpace! 

I'm still keen on this. And interested in both linear and more complex reranking. 

In my head though, linear reranking would run alongside the vector search (biasing the similarity score) while reranking happens post vector search. So of far higher value to me is linear reranking given that I can run my own reranking models directly in node js.

**Comment by zereraz** [comment]
Can't `xenova/transformers.js` be used to support reranking in nodejs?
There is an example here [cross-encoder](https://github.com/xenova/transformers.js/blob/7f5081da29c3f77ee830269ab801344776e61bcb/examples/cross-encoder/src/worker.js#L31)
